ibm/service/resourcecontroller/resource_ibm_resource_key.go

```diff
@@ -298,7 +298,9 @@ func resourceIBMResourceKeyRead(d *schema.ResourceData, meta interface{}) error
 	}
 	d.Set("name", *resourceKey.Name)
 	d.Set("status", *resourceKey.State)
-	if resourceKey.Credentials != nil && resourceKey.Credentials.IamRoleCRN != nil {
+	if resourceKey.Credentials != nil && resourceKey.Credentials.Redacted != nil {
+		log.Printf("Credentials are redacted with code: %s.The User doesn't have the correct access to view the credentials. Refer to the API documentation for additional details.", *resourceKey.Credentials.Redacted)
+	} else if resourceKey.Credentials != nil && resourceKey.Credentials.IamRoleCRN != nil {
```

Comment:
I don't think this else if block get executed if "if" condiiton is satisfied ...make it two seperate if only

Reply:
Done, thank you